BUG: assert_frame_equal still checks category dtypes even when asked not to check index type

### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.


### Reproducible Example

```python
import numpy as np
import pandas as pd
df = pd.DataFrame({"x": [1, 2]})
df.index = pd.MultiIndex.from_arrays(
    [pd.Categorical(np.array([1, 2], dtype=np.uint64))]
)
df2.index = pd.MultiIndex.from_arrays(
    [pd.Categorical(np.array([1, 2], dtype=np.int64))]
)
pd.testing.assert_frame_equal(df, df2, check_categorical=False, check_index_type=False)
```


### Issue Description

When testing two indices and I ask to not check index type I don't think it should check the type of categories of the of type of an index level either.

Note that for a normal index things behave better:

```python
df = pd.DataFrame({"x": [1, 2]})
df.index = pd.Index(pd.Categorical(np.array([1, 2], dtype=np.int64)))
df2.index = pd.Index(pd.Categorical(np.array([1, 2], dtype=np.uint64)))
pd.testing.assert_frame_equal(df, df2, check_categorical=False)
```

Passes as I'd expect.



### Expected Behavior

```python
df = pd.DataFrame({"x": [1, 2]})
df.index = pd.MultiIndex.from_arrays(
    [pd.Categorical(np.array([1, 2], dtype=np.uint64))]
)
df2.index = pd.MultiIndex.from_arrays(
    [pd.Categorical(np.array([1, 2], dtype=np.int64))]
)
pd.testing.assert_frame_equal(df, df2, check_categorical=False, check_index_type=False)
```

Expected behavior: passes. Nothing is printed.

Actual behavior:
```python
AssertionError                            Traceback (most recent call last)
Cell In[163], line 8
      2 df.index = pd.MultiIndex.from_arrays(
      3     [pd.Categorical(np.array([1, 2], dtype=np.uint64))]
      4 )
      5 df2.index = pd.MultiIndex.from_arrays(
      6     [pd.Categorical(np.array([1, 2], dtype=np.int64))]
      7 )
----> 8 pd.testing.assert_frame_equal(df, df2, check_categorical=False, check_index_type=False)

    [... skipping hidden 3 frame]

File ~/.local/lib/python3.10/site-packages/pandas/_testing/asserters.py:479, in assert_categorical_equal(left, right, check_dtype, check_category_order, obj)
    476     exact = True
    478 if check_category_order:
--> 479     assert_index_equal(
    480         left.categories, right.categories, obj=f"{obj}.categories", exact=exact
    481     )
    482     assert_numpy_array_equal(
    483         left.codes, right.codes, check_dtype=check_dtype, obj=f"{obj}.codes"
    484     )
    485 else:

    [... skipping hidden 1 frame]

File ~/.local/lib/python3.10/site-packages/pandas/_testing/asserters.py:247, in assert_index_equal.<locals>._check_types(left, right, obj)
    244         assert_index_equal(left.categories, right.categories, exact=exact)
    245     return
--> 247 assert_attr_equal("dtype", left, right, obj=obj)

    [... skipping hidden 1 frame]

File ~/.local/lib/python3.10/site-packages/pandas/_testing/asserters.py:595, in raise_assert_detail(obj, message, left, right, diff, first_diff, index_values)
    592 if first_diff is not None:
    593     msg += f"\n{first_diff}"
--> 595 raise AssertionError(msg)

AssertionError: MultiIndex level [0] category.categories are different

Attribute "dtype" are different
[left]:  uint64
[right]: int64
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 5c155883fdc5059ee7e21b20604a021d3aa92b01
python           : 3.10.9.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.19.11-1rodete1-amd64
Version          : #1 SMP PREEMPT_DYNAMIC Debian 5.19.11-1rodete1 (2022-10-31)
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 2.1.0.dev0+284.g5c155883fd
numpy            : 1.21.5
pytz             : 2022.6
dateutil         : 2.8.2
setuptools       : 65.6.3
pip              : 22.3.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.9.1
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.10.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : None
brotli           : 1.0.9
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
zstandard        : None
tzdata           : 2022.7
qtpy             : None
pyqt5            : None

</details>

--- a/doc/source/whatsnew/v2.1.0.rst
+++ b/doc/source/whatsnew/v2.1.0.rst
@@ -259,6 +259,7 @@ Other
 ^^^^^
 - Bug in :func:`assert_almost_equal` now throwing assertion error for two unequal sets (:issue:`51727`)
 - Bug in :meth:`Series.memory_usage` when ``deep=True`` throw an error with Series of objects and the returned value is incorrect, as it does not take into account GC corrections (:issue:`51858`)
+- Bug in :func:`assert_frame_equal` checks category dtypes even when asked not to check index type (:issue:`52126`)
 
 .. ***DO NOT USE THIS SECTION***
 

--- a/pandas/_testing/asserters.py
+++ b/pandas/_testing/asserters.py
@@ -294,6 +294,7 @@ def assert_index_equal(
                 exact=exact,
                 check_names=check_names,
                 check_exact=check_exact,
+                check_categorical=check_categorical,
                 rtol=rtol,
                 atol=atol,
                 obj=lobj,

--- a/pandas/tests/util/test_assert_index_equal.py
+++ b/pandas/tests/util/test_assert_index_equal.py
@@ -298,3 +298,16 @@ def test_assert_ea_index_equal_non_matching_na(check_names, check_categorical):
         tm.assert_index_equal(
             idx1, idx2, check_names=check_names, check_categorical=check_categorical
         )
+
+
+@pytest.mark.parametrize("check_categorical", [True, False])
+def test_assert_multi_index_dtype_check_categorical(check_categorical):
+    idx1 = MultiIndex.from_arrays([Categorical(np.array([1, 2], dtype=np.uint64))])
+    idx2 = MultiIndex.from_arrays([Categorical(np.array([1, 2], dtype=np.int64))])
+    if check_categorical:
+        with pytest.raises(
+            AssertionError, match=r"^MultiIndex level \[0\] are different"
+        ):
+            tm.assert_index_equal(idx1, idx2, check_categorical=check_categorical)
+    else:
+        tm.assert_index_equal(idx1, idx2, check_categorical=check_categorical)

--- a/pandas/tests/util/test_assert_index_equal.py
+++ b/pandas/tests/util/test_assert_index_equal.py
@@ -302,6 +302,7 @@ def test_assert_ea_index_equal_non_matching_na(check_names, check_categorical):
 
 @pytest.mark.parametrize("check_categorical", [True, False])
 def test_assert_multi_index_dtype_check_categorical(check_categorical):
+    # GH#52126
     idx1 = MultiIndex.from_arrays([Categorical(np.array([1, 2], dtype=np.uint64))])
     idx2 = MultiIndex.from_arrays([Categorical(np.array([1, 2], dtype=np.int64))])
     if check_categorical:

